scene/2d/tile_map.h

```diff
@@ -215,6 +215,14 @@ class TileMap : public Node2D {
 	LocalVector<TileMapLayer> layers;
 	int selected_layer = -1;
 
+	// Navigation
+	struct TileMapNavigationLayer {
+		String name;
+		RID navigation_map;
+		bool uses_default_map = false;
```

Comment:
You might want to avoid using a flag to determine if the map is the world's default map, since a user could use `set_navigation_map` to assign the world's map to multiple layers, which may then be freed in `_navigation_cleanup_layer`.  Instead, I'd actually compare the RID of the navigation map to the one in the world to determine if you need to free it.

Reply:
I was not sure if the TileMap cleanup layer functions would only ever be call while in the SceneTree. The issue is that  get_world_2d().get_navigation_map() crashes when called while not in the SceneTree so I went with this compromise to track the default map.